src/huggingface_hub/hf_api.py

```diff
@@ -235,7 +236,15 @@ def create_repo(
         if exist_ok and r.status_code == 409:
             d = r.json()
             return d["url"]
-        r.raise_for_status()
+
+        try:
+            r.raise_for_status()
+        except HTTPError as e:
+            if r.text:
+                raise HTTPError("{} - {}".format(e, r.json()["error"]))
```

Comment:
what if `r.text` but not `r.json()` nor `r.json()["error"]`?

shouldn't we test for the more restrictive condition in the if clause?

Reply:
Oops, sorry about this - the `r.text` was a leftover of what I was originally trying.

I updated as you suggested, let me know if you have other suggestions :)